public/js/reducers/tabs.js

```diff
@@ -47,4 +60,5 @@ function getTabId(tab) {
   return id;
 }
 
+const update = setupReducer(initialState, { addTabs, selectTab });
```

Comment:
Lastly, after a little bit of research I found that the redux docs have this helper function [createReducers](https://github.com/reactjs/redux/blob/master/docs/recipes/ReducingBoilerplate.md#generating-reducers) under a section of reducing boilerplate. Given that this is widely documented, I'd opt for using it.